File: Desarrollo Movil/Tarea 1/Punto 2/solucion2/src/Solucion2.java

//------------------------IMPORTACIONES---------------------------
import java.util.Scanner;
//------------------------ZONA DE CÓDIGO------------------------

/**
 * Clase Principal de la Solución:
 * Punto 1: se desea saber cuantos meses han transcurrido entre enero de 1978 y enero de 2020
 */
public class Solucion2 {
    
    /**
     * Método que solicita al usuario la edad de Juan.
     *
     * @param scanner     El objeto Scanner para leer la entrada del usuario.
     * @param maxIntentos El número máximo de intentos para ingresar la edad.
     * @return La edad de Juan ingresada por el usuario.
     */
    public static int recibirEdad(Scanner scanner, int maxIntentos) {
        for (int intento = 0; intento < maxIntentos; intento++) {
            System.out.print("Por favor, ingresa la edad de Juan: ");
            String edadStr = scanner.nextLine();
    
            try {
                int edad = Integer.parseInt(edadStr);
                return edad;
            } catch (NumberFormatException e) {
                System.out.println("Error: Debes ingresar un número entero válido.");
            }
        }
    
        System.out.println("Has alcanzado el número máximo de intentos.");
        System.exit(1);
        return 0;
    }
    
    /**
     * Método que calcula la edad de la Mama de Juan.
     *
     * @param edadJuan La edad de Juan.
     * @param alberto  La proporción de la edad de Alberto con respecto a Juan.
     * @param ana      La proporción de la edad de Ana con respecto a Juan.
     * @return La edad de la Mama de Juan.
     */
    public static double calcularEdadMamaJuan(int edadJuan, double alberto, double ana) {
        return (alberto * edadJuan) + (ana * edadJuan) + edadJuan;
    }

        /**
     * Método principal del programa.
     * @param args Los argumentos de línea de comandos (no se utilizan en este programa).
     */
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int maxIntentos = 3; // Número máximo de intentos para ingresar la edad de Juan
        double alberto = 2.0 / 3.0; // Proporción de la edad de Alberto con respecto a Juan
        double ana = 4.0 / 3.0; // Proporción de la edad de Ana con respecto a Juan
    
        for (int contador = 0; contador < 5; contador++) {
            int edadJuan = recibirEdad(scanner, maxIntentos);
            double edadMamaJuan = calcularEdadMamaJuan(edadJuan, alberto, ana);
            double edadAlberto = alberto * edadJuan;
            double edadAna = ana * edadJuan;
    
            System.out.println("Las edades de la familia de Juan son:");
            System.out.println("Edad de Juan: " + edadJuan);
            System.out.println("Edad de Alberto: " + edadAlberto);
            System.out.println("Edad de Ana: " + edadAna);
            System.out.println("Edad de la Mama de Juan: " + edadMamaJuan);
        }
    }
}

//  appEnd

File: Desarrollo Movil/Tarea 1/Punto 2/solucion2.py
""" Resolución Punto 2: 
    a la mama de juan le preguntan su edad, esta contesta: “tengo 3 hijos, 
    a juan pregúntale su edad, Alberto tiene 2/3 de la edad de juan. 
    Ana tiene 4/3 de la edad de juan, mi edad es la suma de los 3” 
    realiza un algoritmo que muestre la edad de los 4.
"""
#------------------------IMPORTACIONES---------------------------
from functools import reduce
#----------------------DECLARACIÓN DE PARÁMETROS---------------------
max_intentos = 3  # Número máximo de intentos
Alberto = 2/3
Ana = 4/3
#------------------------ZONA DE CÓDIGO------------------------
""" Este Método solicita el valor correspondiente a la edad de Juan y verifica 
        return:
            edad: edad de juan indicada por consola, msg de error de lo contrario
"""
def recepcionEdad():
    for intento in range(max_intentos):
        # Entrada de un valor por consola
        edadStr = input("Por favor, ingresa la edad de Juan:")

        # Verificar si la entrada es un número entero válido
        if edadStr.isdigit():
            edad = int(edadStr)
            break  # Salir del bucle si se ingresa un número válido
        else:
            print("Error: Debes ingresar un número entero válido.")

    else:
        print("Has alcanzado el número máximo de intentos.")
        exit()  # Salir del programa si se alcanza el número máximo de intentos

    return edad

""" Este método calcula las edades de los diferentes miembros de la familia de juan
        parámetros: 
            edadJuan: la edad de juan indicada por el usuario mediante consola
        return:
            msg: mensaje con el resultado.
"""
def edadFamiliaJuan(edadJuan):
    edadMamaJuan = (Alberto * edadJuan)+(Ana * edadJuan)+(edadJuan)
    edadAlberto = Alberto * edadJuan
    edadAna = Ana * edadJuan
    msg = f"Las edades de la familia de Juan son: \n Edad de Juan: {edadJuan} \n Edad de Alberto: {edadAlberto} \n Edad de Ana: {edadAna} \n Edad de la Mama de Juan: {edadMamaJuan}"
    return msg
#-------------------------ZONA DE TEST 1-------------------------
"""función encargada de repetir el algoritmo para hacer varios test
"""
def repTest():
    estado = True
    contador = 0
    while estado == True:
        edad_juan = recepcionEdad()  # Llamamos a la función para obtener la edad de Juan
        print(edadFamiliaJuan(edad_juan))
        contador = contador + 1
        if contador == 5:
            estado = False

repTest()

#   appEnd
